armi/physics/neutronics/const.py

```diff
@@ -52,3 +55,28 @@
 RESTARTFILES = "Restart files"
 INPUTOUTPUT = "Input/Output"
 FLUXFILES = "Flux files"
+
+
+class SolutionType(Enum):
+    """Type of flux computed."""
+
+    REAL = 0
+    ADJOINT = 1
+    REAL_AND_ADJOINT = 2
+
+    @staticmethod
+    def fromGlobalFluxOpts(opts):
+        """Infer solution type from GlobalFluxOptions object."""
+        if opts.real and opts.adjoint:
+            return SolutionType.REAL_AND_ADJOINT
+        elif opts.real:
+            return SolutionType.REAL
+        else:
+            return SolutionType.ADJOINT
```

Comment:
Flags would make this easier, too. `if opts.real: typ |= REAL`, `if opts.adjoint: typ |= ADJOINT`, no special case for both